integration/package-operator/hypershift_test.go

```diff
@@ -106,26 +107,96 @@ func TestHyperShift(t *testing.T) {
-			},
-			Spec: corev1alpha1.PackageSpec{
-				Image:     pkgImage,
-				Component: "hosted-cluster",
-			},
+		// Try to get existing hosted-cluster package
+		hcPkg := &corev1alpha1.Package{}
+		err := Client.Get(ctx, client.ObjectKey{
+			Name:      "hosted-cluster",
+			Namespace: namespace,
+		}, hcPkg)
+
+		if apierrors.IsNotFound(err) {
+			// Package doesn't exist, create it
+			hcPkg = &corev1alpha1.Package{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "hosted-cluster",
+					Namespace: namespace,
+				},
+				Spec: corev1alpha1.PackageSpec{
+					Image:     pkgImage,
+					Component: "hosted-cluster",
+				},
+			}
+			require.NoError(t, Client.Create(ctx, hcPkg))
+		} else {
+			require.NoError(t, err)
 		}
 
-		require.NoError(t, Client.Create(ctx, hcPkg))
-
 		// Wait for roll-out of hosted cluster package
 		err = Waiter.WaitForCondition(
 			ctx, hcPkg, corev1alpha1.PackageAvailable,
 			metav1.ConditionTrue, wait.WithTimeout(100*time.Second),
 		)
 		require.NoError(t, err)
 	})
+
+	t.Run("NodeSelectorPropagation", func(t *testing.T) {
+		// Use the nodeSelector as defined in cmd/build/cluster.go
+		testNodeSelector := map[string]string{
+			"hosted-cluster": "node-selector",
+		}
+
+		// Verify remote-phase deployment has the nodeSelector
+		rpDeployment := &appsv1.Deployment{}
+		require.NoError(t, Client.Get(ctx,
+			client.ObjectKey{
+				Name:      "package-operator-remote-phase-manager",
+				Namespace: namespace,
+			},
+			rpDeployment,
+		))
+		assert.Equal(t, testNodeSelector, rpDeployment.Spec.Template.Spec.NodeSelector,
+			"remote-phase deployment should have the correct nodeSelector")
+
+		// Verify hosted-cluster deployment has the nodeSelector
+		hcDeployment := &appsv1.Deployment{}
+		require.NoError(t, Client.Get(ctx,
+			client.ObjectKey{
+				Name:      "package-operator-hosted-cluster-manager",
+				Namespace: namespace,
+			},
+			hcDeployment,
+		))
+		assert.Equal(t, testNodeSelector, hcDeployment.Spec.Template.Spec.NodeSelector,
+			"hosted-cluster deployment should have the correct nodeSelector")
+
+		// Verify pods are scheduled on nodes matching the selector
+		pods := &corev1.PodList{}
+		err = Client.List(ctx, pods,
+			client.InNamespace(namespace),
+			client.MatchingLabels(map[string]string{
+				"app.kubernetes.io/name": "package-operator",
+			}))
+		require.NoError(t, err)
+
+		for _, pod := range pods.Items {
+			// Verify pod has the nodeSelector
+			assert.Equal(t, testNodeSelector, pod.Spec.NodeSelector,
+				"Pod %s should have the correct nodeSelector", pod.Name)
+
+			// Get the node this pod is scheduled on
+			node := &corev1.Node{}
+			require.NoError(t, Client.Get(ctx,
+				client.ObjectKey{Name: pod.Spec.NodeName},
+				node))
+
+			// Verify node has matching labels
+			for k, v := range testNodeSelector {
+				assert.Equal(t, v, node.Labels[k],
+					"Pod %s/%s scheduled on node %s missing required label %s=%s",
+					pod.Namespace, pod.Name, node.Name, k, v)
+			}
+		}
```

Comment:
Verifying that the deployment has the correct selector in .spec.template.spec.nodeSelector should be enough when coupled with a `Waiter.WaitForCondition(ctx, hcDeployment, appsv1.DeploymentAvailable, metav1.ConditionTrue, wait.WithTimeout(300*time.Second))`.

```suggestion
```